conda_rpms/install.py

```diff
@@ -516,18 +710,55 @@ def link(pkgs_dir, prefix, dist, linktype=LINK_HARD, index=None, target_prefix=N
         return
 
     source_dir = join(pkgs_dir, dist)
-    if not run_script(dist, 'pre-link', prefix, target_prefix):
+    if not run_script(prefix, dist, 'pre-link', target_prefix):
         sys.exit('Error: pre-link failed: %s' % dist)
 
     info_dir = join(source_dir, 'info')
     files = list(yield_lines(join(info_dir, 'files')))
     has_prefix_files = read_has_prefix(join(info_dir, 'has_prefix'))
     no_link = read_no_link(info_dir)
 
+    noarch = False
+    # If the distribution is noarch, it will contain a `link.json` file in
+    # the info_dir
+    with open(join(info_dir, 'index.json'), 'r') as fh:
+        index_data = json.loads(fh.read())
+    if 'noarch' in index_data:
+        noarch = index_data['noarch']
```

Comment:
See https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#architecture-independent-packages

It is possible that `noarch` could be set to `python` or `generic`. I don't think that we'd care about `generic`, but you never know.

Also, do we support the legacy `noarch_python: True` recipe YAML ... if so, then we should have:

```python
if 'noarch' in index_data:
    noarch = index_data['noarch']
elif 'noarch_python' in index_data:
    noarch = index_data['noarch_python']

if noarch:
    if on_win:
        raise ValueError(...)

    ....
```